vj4/handler/judge.py

```diff
@@ -44,6 +50,36 @@ class JudgeDataListView(base.Handler):
       datalist.append({'domain_id': did, 'pid': pid})
     self.json({'list': datalist, 'time': int(time.time())})
 
+@app.route('/judge/data/{rid}', 'data_detail')
+class JudgeDataDetailView(base.Handler):
+  # @base.require_priv(builtin.PRIV_READ_RECORD_CODE | builtin.PRIV_WRITE_RECORD)
+  @base.route_argument
+  @base.sanitize
+  async def get(self, *, rid: objectid.ObjectId):
```

Comment:
This interface is used for serving two kind of pretest data:
- User constructed pretest data
- User uploaded zip, containing pretest data

Currently only the first one is implemented.

Besides, libjudge only accepts zip files (in zipfs), thus it looks like the only difference between "a new protocol" and current implementation is minimal: test case configuration.